docs-crawler/docs/docs/spiders/torch_spider.py

```diff
@@ -6,11 +6,41 @@
 
 class TorchSpider(scrapy.Spider):
     name = "torch"
+    version = "1.4.0"
     split_def = re.compile('^([\w\.]+)\(([\w\,\s=\*\.]*)\)')
 
     def start_requests(self):
         urls = [
-            'https://pytorch.org/docs/stable/torch.html',
+            f'https://pytorch.org/docs/{self.version}/torch.html',
+            f'https://pytorch.org/docs/{self.version}/nn.html',
+            f'https://pytorch.org/docs/{self.version}/nn.functional.html',
+            f'https://pytorch.org/docs/{self.version}/tensors.html',
+            f'https://pytorch.org/docs/{self.version}/tensor_attributes.html',
+            f'https://pytorch.org/docs/{self.version}/autograd.html',
+            f'https://pytorch.org/docs/{self.version}/cuda.html',
+            f'https://pytorch.org/docs/{self.version}/distributed.html',
+            f'https://pytorch.org/docs/{self.version}/distributions.html',
+            f'https://pytorch.org/docs/{self.version}/hub.html',
+            f'https://pytorch.org/docs/{self.version}/jit.html',
+            f'https://pytorch.org/docs/{self.version}/nn.init.html',
+            f'https://pytorch.org/docs/{self.version}/onnx.html',
+            f'https://pytorch.org/docs/{self.version}/optim.html',
+            f'https://pytorch.org/docs/{self.version}/quantization.html',
+            f'https://pytorch.org/docs/{self.version}/rpc.html',
+            f'https://pytorch.org/docs/{self.version}/random.html',
+            f'https://pytorch.org/docs/{self.version}/sparse.html',
+            f'https://pytorch.org/docs/{self.version}/storage.html',
+            f'https://pytorch.org/docs/{self.version}/bottleneck.html',
+            f'https://pytorch.org/docs/{self.version}/checkpoint.html',
+            f'https://pytorch.org/docs/{self.version}/cpp_extension.html',
+            f'https://pytorch.org/docs/{self.version}/data.html',
+            f'https://pytorch.org/docs/{self.version}/dlpack.html',
+            f'https://pytorch.org/docs/{self.version}/model_zoo.html',
+            f'https://pytorch.org/docs/{self.version}/tensorboard.html',
+            f'https://pytorch.org/docs/{self.version}/type_info.html',
+            f'https://pytorch.org/docs/{self.version}/named_tensor.html',
+            f'https://pytorch.org/docs/{self.version}/name_inference.html',
+            f'https://pytorch.org/docs/{self.version}/__config__.html',
```

Comment:
Since these links all exist under a single div in the page we are crawling, we could actually turn this into a [`CrawlSpider`](https://docs.scrapy.org/en/latest/topics/spiders.html#crawlspider) which allows us to use Rules to determine which sub-pages we want to crawl.

We are already using rules in our [tensorflow python spider](https://github.com/OpenMined/threepio/blob/master/docs-crawler/docs/docs/spiders/tf_spider.py#L16-L21) so you could probably use that as an example. I suspect that the CSS selector will need to be changed to something like `div.toctree-wrapper:nth-child(5) a` in order to select the right links.

Reply:
Ahaaa I see, You are right. I didn't notice that, sorry my bad! First time to use `scrapy` 😅
I will modify the code now and push the updates 🙌

Reply:
No worries! :)